#### Suggestions and actions by group

The main objective of this notebook is to provide suggestive actions to the different groups of customers.

In [2]:
import pandas as pd

df_clusters = pd.read_csv(r'C:\Users\danie\Documents\GitHub\customer-loyalty-segmentation\data\clusters.csv')

In [3]:
df_clusters

Customer ID  Recency  Frequency  Monetary  Total Orders  Average Price  \
0           12346    164.0       33.0    206.36          33.0       6.253333   
1           12608     39.0       12.0     22.40          12.0       1.866667   
2           12745    121.0       15.0     27.12          15.0       1.808000   
3           12746    175.0       17.0     59.10          17.0       3.476471   
4           12747      4.0      135.0    485.46         135.0       3.596000   
...           ...      ...        ...       ...           ...            ...   
3833        18283     17.0      211.0    417.22         211.0       1.977346   
3834        18284     66.0       22.0     38.56          22.0       1.752727   
3835        18285    295.0        8.0     24.20           8.0       3.025000   
3836        18286    111.0       59.0    161.89          59.0       2.743898   
3837        18287     17.0       68.0    149.91          68.0       2.204559   

      Cluster  
0           0  
1           1  
2           1  
3           0  
4           1  
...       ...  
3833        1  
3834        1  
3835        0  
3836        1  
3837        1  

[3838 rows x 7 columns]

In [4]:
df_clusters.groupby('Cluster').agg('mean')

Customer ID     Recency    Frequency     Monetary  Total Orders  \
Cluster                                                                     
0        15626.688749  240.789695    26.210305    71.203463     26.210305   
1        15543.757197   43.518939    66.131439   163.619136     66.131439   
2        15595.941667   19.787500   439.762500  1065.004008    439.762500   
3        15633.142857    5.714286  2631.428571  6620.175714   2631.428571   

         Average Price  
Cluster                 
0             2.885710  
1             2.537033  
2             2.429447  
3             2.592238

In [7]:
df_clusters.groupby('Cluster').agg('median')

Customer ID  Recency  Frequency  Monetary  Total Orders  \
Cluster                                                            
0            15613.0    231.0       17.0    45.960          17.0   
1            15573.0     35.0       48.0   117.435          48.0   
2            15732.5     11.0      371.5   886.945         371.5   
3            15311.0      0.0     2364.0  5424.060        2364.0   

         Average Price  
Cluster                 
0             2.715625  
1             2.435875  
2             2.377212  
3             2.439234

#### Recency
- Cluster 3: The lowest recency value is find in this cluster, meaning that customers in this cluster bought recently. Probably, they are **active customers**

- Cluster 0: Have the highest recency value (240 days in average). This indicates that customers in this cluster are **inactive customers**

#### Frequency
- Cluster 3: Apresent the highest frequency value (2631 times in average), indicating that customers in this cluster are **loyal customers**

- Cluster 0: Have the lowes frequency value (26 times in average), indicating that customers in this cluster are **ocasional customers**

#### Monetary
- Cluster 3: Have the highest monetary value (6620 euros in average), indicating that customers in this cluster are **high value customers**

- Cluster 0: Have the lowest monetary value (117 euros in average), suggesting that customers spend less money.


#### Total Orders:
- Cluster 3: Have the highest total orders value (2631 times in average), indicating that customers in this cluster are **big buyers**

- Cluster 0: With fewer orders (26 times in average), reforcing the idea of **inactive customers** or **occasional customers**

#### Average Price:
- Small variations comparing the clusters.

## Conclusion

##### Cluster 0: **Inactive clients** with low frequency and low moneraty spending.
Suggestion: Reactivation campaigns with discounts or special offers.

##### Cluster 1: **Intermediate Clients**
Suggestion: Fidelity programs to increase engagement and total spending.

##### Cluster 2: **Recurrent Clients**
Suggestion: Focus on cross-selling and loyalty programs to increase total spending.

##### Cluster 3: **Highly Valued Clients**:
Suggestion: VIP programs, exclusive offers and reward programs and customized services to maintain the loyalty of these customers.